src/uct/base/uct_iface.c

```diff
@@ -548,3 +549,42 @@ ucs_config_field_t uct_iface_config_table[] = {
 
   {NULL}
 };
+
+ucs_status_t uct_base_ep_am_short_iov(uct_ep_h ep, uint8_t id, const uct_iov_t *iov,
+                                      size_t iovcnt)
+{
+    unsigned length;
+    void *buffer;
+    size_t iov_it;
+    uint64_t header;
+    void *payload;
+    ucs_status_t status;
+
+    length = uct_iov_total_length(iov, iovcnt);
+    if (length > 1200) {
+        buffer = malloc(length);
+    } else {
+        buffer = ucs_alloca(length);
+    }
+    length = 0;
+
+    for (iov_it = 0; iov_it < iovcnt; ++iov_it) {
+        memcpy((char*)buffer + length, iov[iov_it].buffer, iov[iov_it].length);
+        length += iov[iov_it].length;
+    }
+
+    header = *(typeof(header)*)buffer;
+    if (length > sizeof(header)) {
+        payload = (char*)buffer + sizeof(header);
+        length  = length - sizeof(header);
+    } else {
+        payload = NULL;
+        length  = 0;
+    }
+
+    status = uct_ep_am_short(ep, id, header, payload, length);
+    if (length > 1200) {
```

Comment:
if the payload was length = [1201, 1208], it would lead to a leak of the intermediate buffer that was malloc'ed
what about saving the total length to separate variable after getting the result from `uct_iov_total_length()`?

Reply:
Yes, agree.